src/lib.rs

```diff
@@ -285,6 +288,25 @@ impl WalkDir {
         self.opts.max_open = n;
         self
     }
+
+    /// Set a compare function.
+    /// If a compare function is set, WalkDir will return all paths in sorted
+    /// order.
+    ///
+    /// ```rust,no-run
+    /// use std::cmp;
+    /// use std::ffi::OsString;
+    /// use walkdir::WalkDir;
+    ///
+    /// fn string_compare(a: &OsString, b: &OsString) -> cmp::Ordering {
+    ///     a.cmp(b)
+    /// }
+    /// WalkDir::new("foo").sort(Some(string_compare));
+    /// ```
+    pub fn sort(mut self, sorter: Option<fn(&OsString,&OsString) -> std::cmp::Ordering>) -> Self {
```

Comment:
Ah, ya, you probably want a `'static` bound to avoid adding a lifetime parameter. Try `<F: FnMut(&OsString, &OsString) -> Ordering + 'static>`. You'll also need to add `'static` to the type in `WalkDirOptions`.